### ISSUE TITLE
[Tacotron2] Mel_output_postnet of Tacotron2 model has requires_grad as False

### ISSUE THREAD

**Issue by pbansal5** [NVIDIA/DeepLearningExamples#458]
Related to PyTorch

I have made a new inference function which takes as input encoder outputs, instead of text (these encoder outputs are also calculated by another function which takes text and outputs encoder_outputs)
The mel returned by tacotron2 model has require grad as false due to which I am not able to calculate the gradients for the input encoder_outputs (sort of like PGD in encoder_outputs). Hope this makes sense

The tacotron2 is in eval mode and have used the inference function instead of forward
